atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraClientFactory.java

```diff
@@ -105,12 +107,18 @@ public CassandraClient create() {
         try {
             return rateLimitClient(instrumentClient(getRawClientWithKeyspaceSet()));
         } catch (Exception e) {
-            String message = String.format(
-                    "Failed to construct client for %s/%s", cassandraServer.proxy(), clientConfig.keyspace());
             if (clientConfig.usingSsl()) {
-                message += " over SSL";
+                throw new ClientCreationFailedException(
+                        "Failed to construct client for {}/{} over SSL",
+                        e,
+                        SafeArg.of("proxy", CassandraLogHelper.host(cassandraServer.proxy())),
+                        UnsafeArg.of("Keyspace", clientConfig.keyspace()));
```

Comment:
nit: lowercase K :) 

Also, we actually do log keyspace as safe elsewhere (and unsafe too...). Discussing internally, we believe this actually can be made safe here (and the example you found).